### PR TITLE
raftstore: in try_batch, we should use the lastest bucket_meta.

### LINKED ISSUE #12578: In try_batch, we should use the latest apply's bucket_meta.
## Description
Now in struct `Apply`, we have `bucket_meta` field. So we should adjust the try_bacth's code to use the latest `bucket_meta` ().

The code have not handled with `bucket_meta` which means we will use the first one's `bucket_meta`.
image

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
Signed-off-by: SpadeA-Tang <u6748471@anu.edu.au>

### What is changed and how it works?

Issue Number: Close #12578

What's Changed:

```commit-message
Update bucket_meta when the newer apply has.
```

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by 5kbpers — APPROVED** [review]

File: components/raftstore/src/store/fsm/apply.rs [hunk 1]
```diff
@@ -2928,6 +2928,10 @@ impl<S: Snapshot> Apply<S> {
         assert_eq!(self.region_id, other.region_id);
         assert_eq!(self.peer_id, other.peer_id);
         if self.entries_size + other.entries_size <= MAX_APPLY_BATCH_SIZE {
+            if other.bucket_meta.is_some() {
```
Comment (BusyJay) [inline components/raftstore/src/store/fsm/apply.rs:2931]: Any test case?

**Review by tonyxuqqi — APPROVED** [review]

**Review by youjiali1995 — APPROVED** [review]

**Pull request merged**